lib/mock-pipe/mock-pipe.ts

```diff
@@ -3,20 +3,37 @@ import { Pipe, PipeTransform, Type } from '@angular/core';
 import { MockOf } from '../common';
 import { pipeResolver } from '../common/reflect';
 
+const cache = new Map<Type<PipeTransform>, Type<MockedPipe<PipeTransform>>>();
+
+export type MockedPipe<T> = T;
+
 export function MockPipes(...pipes: Array<Type<PipeTransform>>): Array<Type<PipeTransform>> {
   return pipes.map((pipe) => MockPipe(pipe, undefined));
 }
 
 const defaultTransform = (...args: any[]): void => undefined;
-export function MockPipe<TPipe extends PipeTransform>(pipe: Type<TPipe>,
-                                                      transform: TPipe['transform'] = defaultTransform): Type<TPipe> {
-  const pipeName = pipeResolver.resolve(pipe).name;
+export function MockPipe<TPipe extends PipeTransform>(
+  pipe: Type<TPipe>,
+  transform: TPipe['transform'] = defaultTransform,
+): Type<MockedPipe<TPipe>> {
+  const cacheHit = cache.get(pipe);
```

Comment:
is it okay to cache pipe? It wasn't cached, and I assume there's a reason for that.

Reply:
They were cached in the past but I noticed an issue when I tried to declare a new implementation for a pipe in another test file (it was around translations and in one case I cared and in another i didn't) where it didn't let me.

Reply:
Aha, I see, good, then I'll write tests for it and remove the pipe cache from the PR.